### PR TITLE
#28 Implemented Client.send

### PR DESCRIPTION
**Pull request by ddolcimascolo**

### CODE REVIEW

**Comment by MichaelBailly** [comment]
:+1:

File: lib/client/Client.js [hunk 1]
```diff
@@ -494,28 +494,21 @@ export default class Client {
...
+    return this._createMessage(message, MailboxRole.DRAFTS, true);
+  }
 
-        return new CreateMessageAck(this, response.created[clientId]);
-      });
+  /**
+   * Sends a message by issuing a _setMessages_ JMAP request.<br />
+   * The _mailboxIds_ and _isDraft_ properties of the given _message_ will be overridden by this method.
+   *
+   * @param message {OutboundMessage} The message to save.
```
Comment (thomas-hilaire) [inline lib/client/Client.js]: The message to send ?

File: lib/client/Client.js [hunk 2]
```diff
@@ -536,6 +529,31 @@ export default class Client {
     return this.updateMessage(id, { mailboxIds: mailboxIds });
   }
 
+  _createMessage(message, role, isDraft) {
```
Comment (thomas-hilaire) [inline lib/client/Client.js]: Why do not set `isDraft` in the calling method instead of taking a boolean there ?

File: test/common/Client.js [hunk 3]
```diff
@@ -2172,4 +2175,174 @@ describe('The Client class', function() {
...
+            id: 'ma294202da',
+            size: 281,
+            threadId: 'ta294202da'
+          });
+
+          done();
+        });
+    });
+
+    it('should reject the promise if response.created does not contains the expected response format', function(done) {
```
Comment (thomas-hilaire) [inline test/common/Client.js]: does not contain

File: lib/client/Client.js (hunk 2 above)
Comment (ddolcimascolo) [inline lib/client/Client.js]: Code factoring.I'm ding the `isDraft` assignation only here and the calling methods are onliners.

**Comment by thomas-hilaire** [comment]
:+1:

**Pull request merged**